fastly/resource_fastly_ngwaf_virtualpatches.go

```diff
@@ -0,0 +1,157 @@
+package fastly
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	gofastly "github.com/fastly/go-fastly/v11/fastly"
+	ws "github.com/fastly/go-fastly/v11/fastly/ngwaf/v1/workspaces/virtualpatches"
+)
+
+func resourceFastlyNGWAFVirtualPatches() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFastlyNGWAFVirtualPatchUpdate,
+		DeleteContext: resourceFastlyNGWAFVirtualPatchDelete,
+		ReadContext:   resourceFastlyNGWAFVirtualPatchRead,
+		UpdateContext: resourceFastlyNGWAFVirtualPatchUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceFastlyNGWAFVirtualPatchImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"action": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User-configured action of the virtual patch. Accepted values are [`block`, `log`]",
```

Comment:
All of the attribute descriptions should match the ones at https://branch-apidocs-console-1533-ngwaf-functional-parity.developer.fastly.com/documentation/reference/api/ngwaf/virtual-patches/ (also in the data source).